plone/namedfile/scaling.py

```diff
@@ -56,6 +56,9 @@ def __init__(self, context, request, **info):
 
         url = self.context.absolute_url()
         extension = self.data.contentType.split('/')[-1].lower()
+        if self.data.contentType in ["image/svg", "image/svg+xml"]:
+            extension = "svg"
+            self.data.contentType = "image/svg+xml"
```

Comment:
why do you replace "image/svg" with "image/svg+xml"?

Reply:
Because when the image is sent to the browser, the browser doesn't know what to do with image/svg, but it can show fine image/svg+xml.

Reply:
Updated this part too. Now only setting extension to "svg" if content type set to "image/svg+xml".